muesli/web/viewsLecture.py

```diff
@@ -89,25 +91,31 @@ def __call__(self):
                 'times': times,
                 'prefs': utils.preferences}
 
-@view_config(route_name='lecture_add_exam', renderer='muesli.web:templates/lecture/add_exam.pt', context=LectureContext, permission='edit')
+@view_config(route_name='lecture_add_exam', renderer='muesli.web:templates/lecture/add_exam.pt', context=AddExamContext, permission='edit')
 class AddExam(object):
     def __init__(self, request):
         self.request = request
         self.db = self.request.db
         self.lecture_id = request.matchdict['lecture_id']
+        self.category = request.matchdict['category']
```

Comment:
awesome!  you already implemented the category extraction here, so you probably don't even have to do anything when restoring LectureContext